src/plugins/intel_npu/src/backend/src/zero_device.cpp

```diff
@@ -182,12 +179,12 @@ ov::SoPtr<ov::IRemoteTensor> ZeroDevice::createRemoteTensor(std::shared_ptr<ov::
                                                             ov::intel_npu::MemType mem_type,
                                                             void* mem) {
     return {std::make_shared<
-        ZeroRemoteTensor>(context, _initStructs, element_type, shape, config, tensor_type, mem_type, mem)};
+        ZeroRemoteTensor>(std::move(context), _initStructs, element_type, shape, config, tensor_type, mem_type, mem)};
```

Comment:
I would rather go with passing const shared_ptr& into createRemoteTensor instead of bumping ref count first and then explicitly moving shared_ptr. Let's just pass a shared_ptr by reference and make a copy in the RemoteTensor ctor whre it's needed instead of in advance. Such design is more canonical and flexible as code evolves.

Reply:
> Such design is more canonical and flexible as code evolves.

Disagree, it's definitely not canonical - smart pointers are designed to work as explicit memory ownership strategy tool, so you make it explicit in interface (not implementation) if you share resource with the user.